README.md
# Always UP2DATE VALORANT Offsets

If you want to get the right offsets quickly, my repository is just right for you.
I always try to put the most important offsets together for you at a glance as soon as I can.

If you have any suggestions, or if you are convinced that an offset is outdated, please feel free to help fix it. :)

My Unknowncheats Profile (BlackMax97): https://www.unknowncheats.me/forum/members/1532774.html
(Just write me there if you have any questions about the offsets)

If you are looking for other offsets, ask via the VALORANT Reversal thread:
https://www.unknowncheats.me/forum/valorant/385792-valorant-reversal-structs-offsets-312.html


changes

(Only Shooter_classes.h | BasePawn_classes.h)

ABasePawn_C
[~] Name: UberGraphFrame - New Offset: 0x1150 - (Before: 0x1140)
[~] Name: Comp_PlayerCharacter_SpawnInvulnerability - New Offset: 0x1158 - (Before: 0x1148)
------ [~] Name: CharacterPortraitMinimapComponent - New Offset: 0x1160 - (Before: 0x1150)
------ [~] Name: ShooterCharacterMinimapComponent - New Offset: 0x1168 - (Before: 0x1158)
[~] Name: ShooterCharacterVisibility1 - New Offset: 0x1170 - (Before: 0x1160)
[~] Name: SplitBodyRotationRope - New Offset: 0x1178 - (Before: 0x1168)
[~] Name: OutlineOverlay1P - New Offset: 0x1180 - (Before: 0x1170)
[~] Name: OutlineCosmetic3P - New Offset: 0x1188 - (Before: 0x1178)
[~] Name: Comp_CharacterFalling - New Offset: 0x1190 - (Before: 0x1180)
[~] Name: CosmeticActionsComponent - New Offset: 0x1198 - (Before: 0x1188)
[~] Name: Comp_Actor_Concussable - New Offset: 0x11a0 - (Before: 0x1190)
[~] Name: UD_FakeShadow_VFX - New Offset: 0x11a8 - (Before: 0x1198)
[~] Name: IgnoreAllyWallPenetration - New Offset: 0x11b0 - (Before: 0x11a0)
[~] Name: MinimapVision - New Offset: 0x11b8 - (Before: 0x11a8)
[~] Name: AIPerceptionStimuliSource - New Offset: 0x11c0 - (Before: 0x11b0)
[~] Name: Outline1P - New Offset: 0x11c8 - (Before: 0x11b8)
[~] Name: Outline3P - New Offset: 0x11d0 - (Before: 0x11c0)
[~] Name: InEditorHotkeyComponent - New Offset: 0x11d8 - (Before: 0x11c8)
[~] Name: BaseMovementSoundComponent - New Offset: 0x11e0 - (Before: 0x11d0)
[~] Name: SplitBodyRotationGround - New Offset: 0x11e8 - (Before: 0x11d8)
[~] Name: ShooterCharacterDeathReaction - New Offset: 0x11f0 - (Before: 0x11e0)
[~] Name: CharacterHudManager - New Offset: 0x11f8 - (Before: 0x11e8)
[~] Name: ShieldDamageSection - New Offset: 0x1200 - (Before: 0x11f0)
[~] Name: HealthDamageSection - New Offset: 0x1208 - (Before: 0x11f8)
[~] Name: AresHUDListener - New Offset: 0x1210 - (Before: 0x1200)
[~] Name: AudBasePawnVOComponent - New Offset: 0x1218 - (Before: 0x1208)
[~] Name: DefaultMaterials - New Offset: 0x1220 - (Before: 0x1210)
[~] Name: AudBaseSFXComponentClass - New Offset: 0x1230 - (Before: 0x1220)
[~] Name: AudBaseSFXComponent - New Offset: 0x1238 - (Before: 0x1228)
[~] Name: Aud_Base_UI_Component - New Offset: 0x1240 - (Before: 0x1230)
[~] Name: SkipPerceptionSystemRegister - New Offset: 0x1248 - (Before: 0x1238)
[~] Name: FXC_ID_BaseDeaf - New Offset: 0x1250 - (Before: 0x1240)


UAresAimToolingTuningAsset
[~] Name: PullToCenterFeatureTuningSetADS - New Offset: 0x1c8 - (Before: 0x1b0)
[~] Name: PullToCenterWeaponTuningSets - New Offset: 0x230 - (Before: 0x200)
[~] Name: bAimToolingTargetFriction - New Offset: 0x280 - (Before: 0x250)
[~] Name: TargetFrictionSharedTuningSet - New Offset: 0x284 - (Before: 0x254)
[~] Name: TargetFrictionFeatureTuningSet - New Offset: 0x288 - (Before: 0x258)
[~] Name: TargetFrictionFeatureTuningSetADS - New Offset: 0x298 - (Before: 0x268)
[~] Name: TargetFrictionWeaponTuningSets - New Offset: 0x2a8 - (Before: 0x278)
[~] Name: bAimToolingAimAccelerationDelay - New Offset: 0x2f8 - (Before: 0x2c8)
[~] Name: AimAccelerationDelaySharedTuningSet - New Offset: 0x2fc - (Before: 0x2cc)
[~] Name: bAimToolingDeceleration - New Offset: 0x304 - (Before: 0x2d4)
[~] Name: DecelerationSharedTuningSet - New Offset: 0x308 - (Before: 0x2d8)
[~] Name: bAimToolingStrafeAssist - New Offset: 0x320 - (Before: 0x2f0)
[~] Name: StrafeAssistSharedTuningSet - New Offset: 0x324 - (Before: 0x2f4)
[~] Name: StrafeAssistFeatureTuningSet - New Offset: 0x328 - (Before: 0x2f8)
[~] Name: StrafeAssistFeatureTuningSetADS - New Offset: 0x338 - (Before: 0x308)
[~] Name: StrafeAssistWeaponTuningSets - New Offset: 0x348 - (Before: 0x318)
[~] Name: bAimToolingFrameOfReference - New Offset: 0x398 - (Before: 0x368)
[~] Name: FrameOfReferenceSharedTuningSet - New Offset: 0x39c - (Before: 0x36c)
[~] Name: FrameOfReferenceFeatureTuningSet - New Offset: 0x3a0 - (Before: 0x370)
[~] Name: FrameOfReferenceFeatureTuningSetADS - New Offset: 0x3b8 - (Before: 0x388)
[~] Name: FrameOfReferenceWeaponTuningSets - New Offset: 0x3d0 - (Before: 0x3a0)
[~] Name: bAimToolingFlashTurning - New Offset: 0x420 - (Before: 0x3f0)
[~] Name: FlashTurningSharedTuningSet - New Offset: 0x424 - (Before: 0x3f4)
[~] Name: NoTargetFalloffSharedTuningSet - New Offset: 0x430 - (Before: 0x400)
[~] Name: NewTargetRampUpSharedTuningSet - New Offset: 0x43c - (Before: 0x40c)

UAresBotDebugComponent
[~] Name: bUseNewCombatAim - New Offset: 0xec - (Before: 0xe9)
[~] Name: bShouldOverrideAimOffset - New Offset: 0xed - (Before: 0xea)
[~] Name: OverrideAimOffsetType - New Offset: 0xee - (Before: 0xeb)

UAresClientGameInstance
[~] Name: WidgetRegistry - New Offset: 0x1d8 - (Before: 0x1d0)
[~] Name: MyGameInstance - New Offset: 0x1e0 - (Before: 0x1d8)
[~] Name: ContentIndexManager - New Offset: 0x1e8 - (Before: 0x1e0)
[~] Name: InventoryInitManager - New Offset: 0x1f0 - (Before: 0x1e8)
[~] Name: AntiAddictionManager - New Offset: 0x1f8 - (Before: 0x1f0)
[~] Name: AntiAddictionSessionManager - New Offset: 0x200 - (Before: 0x1f8)
[~] Name: CNAntiAddictionManager - New Offset: 0x208 - (Before: 0x200)
[~] Name: VoiceManager - New Offset: 0x210 - (Before: 0x208)
[~] Name: ChatFriendsMetadataManager - New Offset: 0x218 - (Before: 0x210)
[~] Name: PresenceManager - New Offset: 0x220 - (Before: 0x218)
[~] Name: PresenceUpdater - New Offset: 0x228 - (Before: 0x220)
[~] Name: TextChatManagerV2 - New Offset: 0x230 - (Before: 0x228)
[~] Name: MutedPatternsManager - New Offset: 0x238 - (Before: 0x230)
[~] Name: ThreadedChatManager - New Offset: 0x240 - (Before: 0x238)
[~] Name: FriendsManager - New Offset: 0x248 - (Before: 0x240)
[~] Name: DisplayNameManager - New Offset: 0x250 - (Before: 0x248)
[~] Name: PlatformInitSettingsManager - New Offset: 0x258 - (Before: 0x250)
[~] Name: RiotProductManager - New Offset: 0x260 - (Before: 0x258)
[~] Name: AresSettingsManager - New Offset: 0x268 - (Before: 0x260)
[~] Name: RoamingSettingsManager - New Offset: 0x270 - (Before: 0x268)

valorant-offsets.h
namespace offsets {
  // General Offsets
  uintptr_t uworld_state          =     0x8C8A640;                    // @image_base 
  uintptr_t uworld_key            =     uworld_state + 0x38;          // @image_base + (uworld_state + 0x38)
  
  uintptr_t gobj_state            =     0x8ABAA80;                    // @image_base  (Outdated - Updating later)
  uintptr_t gobj_key              =     gobj_state + 0x38;            // @image_base + (gobj_state + 0x38)
  
  uintptr_t fname_pool            =     0x8AC0D80;                    // @image_base
  uintptr_t fname_pool_state      =     0x8AF3D40;                    // @image_base
  uintptr_t fname_pool_key        =     fname_pool_state + 0x38;      // @image_base
  uintptr_t fname_pool_decrypt_w  =     0x313DD20;                    // @image_base
  uintptr_t fname_pool_decrypt_nw =     0x313DCB0;                    // @image_base
  
  uintptr_t find_object           =     0x33B13D0;                    // @image_base
  uintptr_t process_event         =     0x339CE10;                    // @image_base
  
  uintptr_t bone_matrix           =     0x4A79BB0;                    // @image_base
  
  uintptr_t line_of_sight         =     0x4AA88A0;                    // @image_base
  uintptr_t skin_changer_decrypt  =     0x23DB3E0;                    // @image_base
  uintptr_t veh_deref             =     0x1AA9D80;                    // deref_pointer_in_game_space_fn - https://www.unknowncheats.me/forum/valorant/503616-dumping-valorant-perfect-results-easy.html

  uintptr_t enemy_outline_color   =     0x8BA4ED0;                    // @image_base (FLinearColor) 
  uintptr_t ally_outline_color    =     0x8BA4940;                    // @image_base (FLinearColor) 
  uintptr_t ares_outline_mode     =     0x4A29730;                    // @image_base
 
  uintptr_t proj_world_to_screen  =     0x4BC7200;                    // @image_base (ProjectWorldToScreen)
  uintptr_t get_actor_bounds      =     0x4856060;                    // @image_base
  
  // VTables
  
  // World
  uintptr_t persistent_level      =     0x38;                  // @uworld (ULevel)
  uintptr_t owning_game_instance  =     0x1A0;                 // @uworld (UGameInstance)    
  uintptr_t game_state            =     0x140;                 // @uworld (AGameStateBase)
  uintptr_t levels                =     0x158;                 // @uworld (TArray<struct ULevel*>)
  
  // Game Instance
  uintptr_t local_players         =     0x40;                  // @game_instance (TArray<struct ULocalPlayer*>)
  
  // Level
  uintptr_t actor_array           =     0xA0;                  // @level
  uintptr_t actor_count           =     actor_array + 0x18;    // @level + (actor_array + 0x18)
  
  // Local Player
  uintptr_t viewport_client       =     0x78;                  // @local_player (UGameViewportClient)
  uintptr_t player_controller     =     0x38;                  // @local_player (APlayerController)
  
  // Player Controller
  uintptr_t acknowledged_pawn     =     0x460;                 // @player_controller (APawn)
  uintptr_t player_camera         =     0x478;                 // @player_controller (APlayerCameraManager)
  uintptr_t control_rotation      =     0x440;                 // @player_controller (FRotator)
  
  // Player Pawn
  uintptr_t root_component        =     0x230;                 // @player_pawn (USceneComponent)
  uintptr_t damage_handler        =     0xA10;                 // @player_pawn (UDamageableComponent)
  uintptr_t actor_id              =     0x18;                  // @player_pawn (int)
  uintptr_t fname_id              =     0x38;                  // @player_pawn (int)
  uintptr_t dormant               =     0x100;                 // @player_pawn (bool)
  uintptr_t player_state          =     0x3F0;                 // @player_pawn (APlayerState)
  uintptr_t current_mesh          =     0x430;                 // @player_pawn (USkeletalMeshComponent)
  uintptr_t inventory             =     0x9B0;                 // @player_pawn (UAresInventory)
  uintptr_t outline_component     =     0x11D0;                // @player_pawn (UAresOutlineComponent)
  uintptr_t portrait_map          =     0x1160;                // @player_pawn (UCharacterPortraitMinimapComponent_C)
  uintptr_t character_map         =     portrait_map + 0x8;    // @player_pawn (UShooterCharacterMinimapComponent_C)
  
  // Inventory
  uintptr_t current_equippable    =     0x248;                 // @inventory (AAresEquippable)
  
  // Minimap Component
  uintptr_t local_observer        =     0x530;                 // @player_pawn (bool)
  uintptr_t is_visible            =     0x501;                 // @player_pawn (bool)
  
  // Mesh
  uintptr_t component_to_world    =     0x250;                 // @mesh (FTransform)
  uintptr_t bone_array            =     0x5c0;                 // @mesh (FBoxSphereBounds)
  uintptr_t bone_count            =     bone_array + 0x8;      // @mesh (int)
  uintptr_t last_sub_time         =     0x378;
  uintptr_t last_render_time      =     last_sub_time + 0x4;   // @mesh + (last_sub_time + 0x4) 
  uintptr_t outline_mode          =     0x330;                 // @mesh (int)
  uintptr_t attach_children       =     0x110;                 // @mesh (TArray<struct UMesh*>)
  uintptr_t attach_children_count =     attach_children + 0x8; // @mesh (int)
  
  // Player State
  uintptr_t team_component        =     0x628;                 // @player_state (UBaseTeamComponent)
  
  // Team Component
  uintptr_t team_id               =     0xF8;                  // @team_component (int)
  
  // Damage Handler               
  uintptr_t current_health        =     0x1B0;                 // @damage_handler (float)
  uintptr_t max_life              =     current_health + 0x4;  // @damange_handler + (current_health + 0x4)

  // Root Component
  uintptr_t relative_location     =     0x164;                 // @root_component (FVector)
  uintptr_t relative_rotation     =     0x170;                 // @root_component (FRotator)
  
  // Camera Manager
  uintptr_t default_fov           =     0x3ec;                 // @camera (float)
  uintptr_t camera_cache          =     0x510;                 // @camera (FCameraCacheEntry)
  
  // FCameraCacheEntry
  uintptr_t pov                   =     0x10;                  // @camera_cache (FMinimalViewInfo)
  
  // FMinimalViewInfo
  uintptr_t location              =     0x0;                   // @view_info + 0x0 = Location (FVector)
  uintptr_t rotation              =     0x0c;                  // @view_info (FRotator)
  uintptr_t current_fov           =     0x18;                  // @view_info (float)

  // Outline Component
  uintptr_t enemy_outline         =     0x2B1;                 // @outline_component (int)
}
